### PR TITLE
Deprecate all symbols in the Doctrine\Common\Proxy namespace

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Allowed by https://github.com/doctrine/orm/pull/10837

### CODE REVIEW

File: src/Proxy/ProxyGenerator.php [hunk 1]
```diff
@@ -259,6 +262,14 @@ public function __getLazyProperties(): array
      */
     public function __construct($proxyDirectory, $proxyNamespace)
     {
+        if (PHP_VERSION_ID >= 80100) {
```
Comment (nicolas-grekas) [inline src/Proxy/ProxyGenerator.php]: version check matching https://github.com/doctrine/orm/pull/10837#discussion_r1261234947

**Comment by derrabus** [comment]
Do we actually plan to release a v4 without Proxies? If the ORM does not use common proxies anymore, we'd simply drop the dependency.

**Comment by nicolas-grekas** [comment]
> Do we actually plan to release a v4 without Proxies?

This would cut the maintenance yes. Proxies are hard to maintain, especially in the form they take in this repo (see PRs related to readonly or enums for examples.)

**Comment by derrabus** [comment]
I understand that, but what remains of this library if we remove proxies is not worth maintaining imho. I think, we can simply archive the repo once the ORM doesn't use it anymore.

**Comment by nicolas-grekas** [comment]
It'd still be great to release a version with deprecations, so that ppl that rely on eg `instanceof Common\Proxy\Proxy` can know they should move on.

**Comment by derrabus** [comment]
Okay, how about:

* Annotate _all_ classes and interfaces with `@deprecated`, not just the Proxy stuff
* Remove all mentions of version 4.0
* No runtime deprecation warnings as there is no urge to resolve them.
* A note in the README about our plans of sunsetting this library

**Comment by nicolas-grekas** [comment]
>Remove all mentions of version 4.0
> No runtime deprecation warnings as there is no urge to resolve them.

Done for those

For the rest, that's a separate effort to me.

File: src/Proxy/Autoloader.php [hunk 2]
```diff
@@ -20,7 +20,7 @@
 /**
  * Special Autoloader for Proxy classes, which are not PSR-0 compliant.
  *
- * @internal
+ * @deprecated The Autoloader class is deprecated since doctrine/common 3.5.
```
Comment (derrabus) [inline src/Proxy/Autoloader.php]: Deprecating the class does not make it not internal. 😬

**Review by derrabus — DISMISSED** [review]

File: src/Proxy/Autoloader.php (hunk 2 above)
Comment (nicolas-grekas) [reply src/Proxy/Autoloader.php]: fix :see_no_evil:

**Review by derrabus — APPROVED** [review]

**Pull request merged**